python/ray/air/tests/test_dataset_config.py

```diff
@@ -398,6 +399,111 @@ def check_error(shard, results):
     test.fit()
 
 
+def test_deterministic_per_epoch_preprocessor(ray_start_4_cpus):
+    ds = ray.data.range_table(5)
+
+    def multiply(x):
+        return x * 2
+
+    for max_object_store_memory_fraction in [None, 1, 0.3]:
```

Comment:
should we use `pytest.mark.parametrize` for this so that it will be easier to identify which case fails?